### ISSUE TITLE
train problem

### ISSUE THREAD

**Issue by buaaswf** [facebookresearch/Detectron#175]
IndexError: list index out of range
WARNING memonger.py:  70: NOTE: Executing monger to optimize gradient memory
I0221 17:03:58.520751   855 memonger.cc:253] Remapping 83 using 19 shared blobs.
INFO memonger.py: 112: Memonger memory optimization took 0.00770401954651 secs
I0221 17:03:58.904412   855 context_gpu.cu:321] GPU 0: 153 MB
I0221 17:03:58.904430   855 context_gpu.cu:325] Total: 153 MB
I0221 17:03:58.922719   855 context_gpu.cu:321] GPU 0: 311 MB
I0221 17:03:58.922729   855 context_gpu.cu:325] Total: 311 MB
INFO train_net.py: 253: Loading dataset: ('coco_2014_train',)
loading annotations into memory...
Done (t=0.50s)
creating index...
index created!
INFO roidb.py:  49: Appending horizontally-flipped training examples...
INFO roidb.py:  51: Loaded dataset: coco_2014_train
INFO roidb.py: 135: Filtered 23632 roidb entries: 23632 -> 0
INFO roidb.py:  67: Computing bounding-box regression targets...
INFO roidb.py:  69: done
Traceback (most recent call last):
  File "tools/train_net.py", line 281, in <module>
    main()
  File "tools/train_net.py", line 119, in main
    checkpoints = train_model()
  File "tools/train_net.py", line 133, in train_model
    setup_model_for_training(model, output_dir)
  File "tools/train_net.py", line 231, in setup_model_for_training
    add_model_training_inputs(model)
  File "tools/train_net.py", line 255, in add_model_training_inputs
    cfg.TRAIN.DATASETS, cfg.TRAIN.PROPOSAL_FILES
  File "/home1/Detectron/lib/datasets/roidb.py", line 71, in combined_roidb_for_training
    _compute_and_log_stats(roidb)
  File "/home1/Detectron/lib/datasets/roidb.py", line 181, in _compute_and_log_stats
    classes = roidb[0]['dataset'].classes
IndexError: list index out of range

**Comment by ir413** [comment]
Hi @buaaswf, please follow the steps and provide the information suggested in the [issue_template](https://github.com/facebookresearch/Detectron/blob/master/.github/issue_template.md).

**Comment by YoungMagic** [comment]
Have you solved the problem? I have run into the same problem when training my custom dataset.

**Comment by YoungMagic** [comment]
I found the solution. It's because my json file didn't include categories node.

**Comment by LightningChan** [comment]
@YoungMagic I have run into the same problem，how do you create your coco-like json?

**Comment by YoungMagic** [comment]
You can download the PASCAL VOA Annotations in COCO format from [here](https://storage.googleapis.com/coco-dataset/external/PASCAL_VOC.zip). Just follow the same format will be alright. And the segmention in annotations should be 
```
segm = [[bndbox[0], bndbox[2], bndbox[0], bndbox[3], bndbox[1], bndbox[3], bndbox[1], bndbox[2], bndbox[0], bndbox[2]]]
```
@LightningChan

**Comment by ycAlex11** [comment]
hi guys
I have got same issue. In my coco-like json,  I have same structure as coco's json file, but I have deleted some useless information: like 'info', 'license'. What I have in my json file are 3 keys: 'images', 'annotations' and 'categories' which have same structure as coco's json file. But at the end I got an empty roidb which cause : classes = roidb[0]['dataset'].classes
                                             IndexError: list index out of range
Can anyone tell me how could it possible be an empty list for roidb !!!!
